fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java

```diff
@@ -42,19 +44,48 @@ public StructLiteral() {
     }
 
     public StructLiteral(List<Literal> fields) {
-        super(computeDataType(fields));
-        this.fields = ImmutableList.copyOf(fields);
+        this(fields, computeDataType(fields));
+    }
+
+    private StructLiteral(List<Literal> fields, DataType dataType) {
+        super(dataType);
+        this.fields = ImmutableList.copyOf(Objects.requireNonNull(fields, "fields should not be null"));
+        Preconditions.checkArgument(dataType instanceof StructType,
+                "dataType should be StructType, but we meet %s", dataType);
+        Preconditions.checkArgument(fields.size() == ((StructType) dataType).getFields().size(),
+                "fields size is not same with dataType size. %s vs %s",
+                fields.size(), ((StructType) dataType).getFields().size());
     }
 
     @Override
     public List<Literal> getValue() {
         return fields;
     }
 
+    @Override
+    protected Expression uncheckedCastTo(DataType targetType) throws AnalysisException {
```

Comment:
Is here any chance call legacy literal uncheckedCastTo ?

Reply:
no